### PR TITLE
Fix component platform navigation for Next.js

### PR DESCRIPTION
**Pull request by danoc**
The platform names are now sorted and have a display title.

### CODE REVIEW

File: next/utils/mdx-get-static-props.ts [hunk 1]
```diff
@@ -32,8 +32,39 @@ export default function getStaticProps(
         componentPlatforms = fs
             .readdirSync(`pages/components/${metadata.component.id}`)
             .map(file => {
-                // Remove `.mdx` from string
-                return file.replace('.mdx', '');
+                // Removes `.mdx` from filename
+                const platformId = file.replace('.mdx', '');
+
+                const displayName: Record<string, string> = {
```
Comment (lavelle) [inline next/utils/mdx-get-static-props.ts:38]: It's stricter to key this as 

```
type PlatformName = 'usage' | 'react' | 'scss'
Record<PlatformName, string>
```

File: next/utils/mdx-get-static-props.ts [hunk 2]
```diff
@@ -32,8 +32,39 @@ export default function getStaticProps(
...
+                    ios: 'iOS (UIKit)',
+                    swiftui: 'iOS (SwiftUI)',
+                    android: 'Android',
+                };
+
+                // Return the filename as `platformId` as well as a display name for the platform.
+                return { id: platformId, name: displayName[platformId] };
+            })
+            .sort((a, b) => {
+                const platformOrder: Record<string, number> = {
```
Comment (lavelle) [inline next/utils/mdx-get-static-props.ts:51]: use that new `PlatformName` type here too

**Review by lavelle — COMMENTED** [review]
Just a small nit

File: next/utils/component-page-props.ts [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { ComponentDefinition } from '../components/thumbprint-components/props-t
 export interface ComponentPageProps {
     id: string;
     platformId: string;
-    componentPlatforms: string[];
+    componentPlatforms: { id: string; name: string }[];
```
Comment (lavelle) [inline next/utils/component-page-props.ts:7]:
```suggestion
    componentPlatforms: { id: PlatformName; name: string }[];
```

File: next/utils/mdx-get-static-props.ts [hunk 4]
```diff
@@ -32,8 +38,43 @@ export default function getStaticProps(
...
+                const displayName: Record<PlatformName, string> = {
+                    usage: 'Usage',
+                    react: 'React',
+                    scss: 'SCSS',
+                    ios: 'iOS (UIKit)',
+                    swiftui: 'iOS (SwiftUI)',
+                    android: 'Android',
+                };
+
+                if (!isPlatformName(platformId)) {
```
Comment (lavelle) [inline next/utils/mdx-get-static-props.ts:53]:
```suggestion
                if (!isPlatformName(platformId) || !displayName[platformId]) {
```

**Review by lavelle — DISMISSED** [review]
Couple more minor improvements to the types, but  not essential - approving to unblock you

**Comment by danoc** [comment]
Going to self merge since all I did was apply your GH suggestions via the UI

**Pull request merged**